docs/content/concepts/metadata-tags/asset-metadata.mdx

```diff
@@ -0,0 +1,293 @@
+
+### Definition metadata
+
+Definition metadata is attached to the asset's definition using the `metadata` argument. In the following example, we attached a list of columns that the table produced by the asset should contain:
+
+```python file=/concepts/assets/asset_definition_metadata.py startafter=start endbefore=end
+from dagster_duckdb import DuckDBResource
+
+from dagster import asset
+
+# ... other assets
+
+
+@asset(
+    deps=[iris_dataset],
+    metadata={
+        "Expected columns": "sepal_length_cm, sepal_width_cm, petal_length_cm, petal_width_cm, species"
+    },
+)
+def small_petals(duckdb: DuckDBResource) -> None:
+    with duckdb.get_connection() as conn:
+        conn.execute(
+            "CREATE TABLE iris.small_petals AS SELECT * FROM iris.iris_dataset WHERE"
+            " 'petal_length_cm' < 1 AND 'petal_width_cm' < 1"
+        )
+```
+
+### Materialization metadata
+
+To add materialization metadata to an asset, you'll need to:
+
+- Return a <PyObject object="MaterializeResult" /> object with the `metadata` parameter
+- Use the <PyObject object="MetadataValue" /> utility class to wrap the data, ensuring it displays correctly in the UI
+
+In the following example, we added a row count and preview to a `topstories` asset:
+
+```python file=/tutorial/building_an_asset_graph/assets_with_metadata.py lines=3-5,13,16-42
+import json
+import requests
+import pandas as pd
+from dagster import AssetExecutionContext, MetadataValue, asset, MaterializeResult
+
+
+@asset(deps=[topstory_ids])
+def topstories(context: AssetExecutionContext) -> MaterializeResult:
+    with open("data/topstory_ids.json", "r") as f:
+        topstory_ids = json.load(f)
+
+    results = []
+    for item_id in topstory_ids:
+        item = requests.get(
+            f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json"
+        ).json()
+        results.append(item)
+
+        if len(results) % 20 == 0:
+            context.log.info(f"Got {len(results)} items so far.")
+
+    df = pd.DataFrame(results)
+    df.to_csv("data/topstories.csv")
+
+    return MaterializeResult(
+        metadata={
+            "num_records": len(df),  # Metadata can be any key-value pair
+            "preview": MetadataValue.md(df.head().to_markdown()),
+            # The `MetadataValue` class has useful static methods to build Metadata
+        }
+    )
+```
+
+---
+
+## Viewing asset metadata in the Dagster UI
+
+Metadata attached to assets shows up in a few places in the [Dagster UI](/concepts/webserver/ui).
+
+<TabGroup>
+<TabItem name="Global asset lineage">
+
+### Global asset lineage
+
+In the [**Global asset lineage**](/concepts/webserver/ui#global-asset-lineage) page, click on an asset to open the asset details in the sidepanel:
+
+<!-- ![Asset details sidepanel showing metadata in the Global asset lineage page of the Dagster UI](/images/concepts/metadata-tags/metadata-asset-lineage.png) -->
```

Comment:
is the `<!-- ... -->` block here intentional? (I see the image below so not sure)

Reply:
It is, unfortunately. It makes re-running `mdx-format` easier when an image is replaced - instead of having to re-do the Markdown, you can just copy/paste it from the comment and run the command again.